magic_filter/magic.py

```diff
@@ -103,23 +67,29 @@ def endswith(self, value: str) -> AttrOperation:
         """
         return AttrOperation(operation=str.endswith, value=value, magic=self)
 
-    def func(self, func: Callable[..., Any]) -> FuncOperation:
+    def func(self, value: Callable[..., Any]) -> FuncOperation:
         """
         Execute any callable on value
         """
-        return FuncOperation(value=func, magic=self)
+        return FuncOperation(value=value, magic=self)
 
-    def all(self, *call: Callable[[Any, DataType], bool]) -> AllFuncOperation:
+    def all(self, *call: Callable[[Any, Any], bool]) -> AllFuncOperation:
         """
         Analog of builtin all(Iterable[Any])
         """
         return AllFuncOperation(value=call, magic=self)
 
-    def any(self, *call: Callable[[Any, DataType], bool]) -> AnyFuncOperation:
+    def any(self, *call: Callable[[Any, Any], bool]) -> AnyFuncOperation:
         """
         Analog of builtin any(Iterable[Any])
         """
         return AnyFuncOperation(value=call, magic=self)
 
+    # PRIVATE API
+    def get_chain(self) -> List[Attribute]:
```

Comment:
What about property without setter?
```suggestion
    @property
    def chain(self) -> List[Attribute]:
```

Reply:
clean namespace. object field may have chain attr. am I right?

Reply:
you're right
hmm...
what do you think about passing a chain instead of a filter instance in an operation?

Reply:
I agree with this one